### PR TITLE
feat: configure minimum TLS version WPB-4260

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4260" title="WPB-4260" target="_blank">TaskWPB-4260</a>  [iOS] TLS 1.3
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This PR contains changes to make it possible to specify a minimum TLS version to be used when communicating with the backend REST API, web socket, and client blacklist. Changes include:
- Bump Starscream to get api to specify min tls version for web sockets
- Add build time var to specify a min TLS version (1.2 or 1.3).
- Pass the min TLS version to url sessions of `ZMTransportSession` (for backend REST API)
- Pass the min TLS version to all websocket implementations
- Pass the min TLS version to the `ZMBlacklistDownloader`

In case no min TLS version is specified, v1.2 is used.

#### How to Test

- Use a build that has a specified min TLS version.
- Connect to a backend that uses a lower TLS version than the min version.
- Assert that communication with the backend is not possible.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — APPROVED** [review]

File: wire-ios-transport/Source/PushChannel/NativePushChannel.swift [hunk 1]
```diff
@@ -58,25 +58,34 @@ class NativePushChannel: NSObject, PushChannelType {
...
+        proxyPassword: String?,
+        minTLSVersion: String?,
+        queue: OperationQueue
+    ) {
         self.environment = environment
         self.scheduler = scheduler
         self.workQueue = queue
         self.proxyUsername = proxyUsername
         self.proxyPassword = proxyPassword
+        self.minTLSVersion = minTLSVersion.flatMap(TLSVersion.init) ?? .v1_2
```
Comment (netbe) [inline wire-ios-transport/Source/PushChannel/NativePushChannel.swift]: What about defining a global constant for the fallback version v1.2 ?

File: wire-ios-transport/Source/PushChannel/StarscreamPushChannel.swift [hunk 2]
```diff
@@ -66,17 +67,21 @@ class StarscreamPushChannel: NSObject, PushChannelType {
...
+        proxyPassword: String?,
+        minTLSVersion: String?,
+        queue: OperationQueue
+    ) {
         self.environment = environment
         self.scheduler = scheduler
         self.proxyUsername = proxyUsername
         self.proxyPassword = proxyPassword
         self.workQueue = queue
+        self.minTLSVersion = minTLSVersion.flatMap(TLSVersion.init) ?? .v1_2
```
Comment (netbe) [inline wire-ios-transport/Source/PushChannel/StarscreamPushChannel.swift]: same goes here

File: wire-ios/Wire-iOS.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1329,6 +1329,8 @@
 		EE948F1923E17778000A663E /* ContactsDataSource.swift in Sources */ = {isa = PBXBuildFile; fileRef = EE948F1823E17778000A663E /* ContactsDataSource.swift */; };
 		EE94CA1925C2E490005F1155 /* ScreenCurtainTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = EE94CA0B25C2E48B005F1155 /* ScreenCurtainTests.swift */; };
 		EE98A09E296ED46D0055E981 /* DeviceConfigurationEventHandler.swift in Sources */ = {isa = PBXBuildFile; fileRef = EE98A09D296ED46D0055E981 /* DeviceConfigurationEventHandler.swift */; };
+		EE98AD0D2AAAF05F005C30AD /* Bundle+SecurityFlags.swift in Sources */ = {isa = PBXBuildFile; fileRef = 54D5E01824E1286B00B982F6 /* Bundle+SecurityFlags.swift */; };
```
Comment (netbe) [inline wire-ios/Wire-iOS.xcodeproj/project.pbxproj]: why added in shared Extension? shouldn't it be in WireComponents?

**Review by netbe — COMMENTED** [review]
just some questions

File: wire-ios-transport/Source/PushChannel/NativePushChannel.swift (hunk 1 above)
Comment (johnxnguyen) [reply wire-ios-transport/Source/PushChannel/NativePushChannel.swift]: So we can configure the fallback just once?

File: wire-ios/Wire-iOS.xcodeproj/project.pbxproj (hunk 3 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS.xcodeproj/project.pbxproj]: It's already a target of Wire-iOS and WireShareExtension. I added it as a target to WireNotificationExtension. I'll try make it a target of only common components.

File: wire-ios-transport/Source/PushChannel/NativePushChannel.swift (hunk 1 above)
Comment (netbe) [reply wire-ios-transport/Source/PushChannel/NativePushChannel.swift]: yes

File: wire-ios/Wire-iOS.xcodeproj/project.pbxproj (hunk 3 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS.xcodeproj/project.pbxproj]: done

File: wire-ios-transport/Source/PushChannel/NativePushChannel.swift (hunk 1 above)
Comment (johnxnguyen) [reply wire-ios-transport/Source/PushChannel/NativePushChannel.swift]: done

File: wire-ios-transport/Source/PushChannel/StarscreamPushChannel.swift (hunk 2 above)
Comment (johnxnguyen) [reply wire-ios-transport/Source/PushChannel/StarscreamPushChannel.swift]: done

**Review by netbe — APPROVED** [review]

**Pull request merged**